File: examples/java-ddd/api/src/main/java/org/myddd/java/api/UserApplication.java
package org.myddd.java.api;

import org.myddd.java.api.dto.UserDTO;
import org.myddd.utils.Page;

public interface UserApplication {

    UserDTO createUser(UserDTO userDTO);

    Page<UserDTO> searchUser(int page,int pageSize,String searchKey);
}

File: examples/kotlin-ddd/api/src/main/kotlin/org/myddd/kotlin/api/UserApplication.kt
package org.myddd.kotlin.api

import org.myddd.kotlin.api.dto.UserDTO
import org.myddd.utils.Page

interface UserApplication {

    fun createUser(userDTO: UserDTO):UserDTO?

    fun pageSearchUser(search: String,pageIndex:Int,pageSize:Int = 50):Page<UserDTO>

}